### PR TITLE
envsubst: add page

### LINKED ISSUE #1862: page request: envsubst
I don't know how to use it myself so can't contribute.

### PR DESCRIPTION
**Pull request by pxgamer**
Closes #1862

----

- [x] The page (if new), does not already exist in the repo.

- [x] The page (if new), has been added to the correct platform folder:  
      `common/` if it's common to all platforms, `linux/` if it's Linux-specific, and so on.

- [x] The page has 8 or fewer examples.

- [x] The PR is appropriately titled:  
      `<command name>: add page` for new pages, or `<command name>: <description of changes>` for pages being edited.

- [x] The page follows the [contributing](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md) guidelines.

### CODE REVIEW

File: pages/linux/envsubst.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# envsubst
+
+> Substitutes shell format strings with environment variables in text.
+> Strings to be replaced should be in either `${var}` or `$var` format.
+
+- Replace environment variables in stdin and output to stdout:
+
+`echo "{{\$HOME}}" | envsubst`
```
Comment (agnivade) [inline pages/linux/envsubst.md]: You can use single quotes to prevent shell from interpreting the variable, hence simplifying to - `echo '{{$HOME}}'`

Comment (pxgamer) [reply pages/linux/envsubst.md]: Updated, not sure why I didn't think of that. :+1:

File: pages/linux/envsubst.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+> Substitutes shell format strings with environment variables in text.
+> Strings to be replaced should be in either `${var}` or `$var` format.
+
+- Replace environment variables in stdin and output to stdout:
+
+`echo '{{$HOME}}' | envsubst`
+
+- Replace environment variables in an input file and output to stdout:
+
+`envsubst < {{path/to/input}}`
```
[the remark this answers is not in the document]
Comment (sbrl) [reply pages/linux/envsubst.md]: Hrm, ok.

**Comment by ahmetb** [comment]
this doesn't show up on my tldr setup even after updating through `--update`. I'm using zsh on macOS. I have the gnu gettext package, so I should be able to use it. Any ideas why it's not showing up on macOS?

**Comment by pxgamer** [comment]
@ahmetb it won't show up on Mac because it's been added to the Linux directory (I didn't add it to `common` as it's not Windows-compatible).

You can either specify `--linux` when running the command, or make a copy for Mac if you'd like. :smile:

**Comment by ahmetb** [comment]
🤔 interesting. Why don’t we create a unix/ for such commands and lookup both unix/ then mac/ in the tldr code?

**Comment by sbrl** [comment]
@ahmetb Depending on your client, it may have the ability to show a page from other platforms. The bash client I use, for example, does it automatically. In other clients there may be a `--platform` option, and in still others you can specify it via `platform/pagename`.

**Pull request merged**